agent/mapcall/src/main/java/randoop/instrument/CallReplacementTransformer.java

```diff
@@ -25,140 +26,76 @@
- * The Instrument class is responsible for modifying another class's bytecode. It changes calls to
- * specified functions into static calls to user functions. This allows a user of Randoop to capture
- * arbitrary calls and modify them as desired. For example, all calls to
- * JOptionPane.showConfirmDialog (which would otherwise hang a regressions test) can be changed to
- * throw an exception, or simply print a message and continue.
+ * The {@code CallReplacementTransformer} replaces each call to method m1 by a call to method m2, as
+ * specified by files loaded with {@link #readMapFile(File)} or {@link #readMapFile(Reader)}. It is
+ * used by the MapCallsAgent.
+ *
+ * @see MapCallsAgent
  */
 public class CallReplacementTransformer implements ClassFileTransformer {
 
-  boolean debug = false;
-  boolean log_on = false;
-  boolean debug_class = false;
-
-  /** current Constant Pool */
-  static ConstantPoolGen pgen = null;
-
-  protected InstructionFactory ifact;
+  /** Current Constant Pool */
+  private static ConstantPoolGen pgen = null;
 
   /** Debug information about which classes are transformed and why */
-  public static SimpleLog debug_transform = new SimpleLog(false);
+  private static SimpleLog debug_transform = new SimpleLog(MapCallsAgent.debug);
 
-  public static SimpleLog debug_instrument = new SimpleLog(false);
-  public static SimpleLog debug_instrument_inst = new SimpleLog(false);
+  private static SimpleLog debug_instrument_inst = new SimpleLog(MapCallsAgent.debug);
 
   /** Debug information on method maping */
-  public static SimpleLog debug_map = new SimpleLog("method_mapping.txt", true);
+  private static SimpleLog debug_map = new SimpleLog("method_mapping.txt", MapCallsAgent.debug);
 
-  /** Class that defines a method (by its fully-qualified name and argument types) */
-  static class MethodDef {
-    /** Fully-qualified method name, such as "javax.swing.JOptionPane.showMessageDialog" */
-    String name;
-    /** The argument types */
-    Type[] arg_types;
-
-    MethodDef(String name, Type[] arg_types) {
-      this.name = name;
-      this.arg_types = arg_types;
-    }
-
-    boolean equals(String name, Type[] arg_types) {
-      if (!name.equals(this.name)) return false;
-      if (this.arg_types.length != arg_types.length) return false;
-      for (int ii = 0; ii < arg_types.length; ii++)
-        if (!arg_types[ii].equals(this.arg_types[ii])) return (false);
-      return (true);
-    }
-
-    @Override
-    public boolean equals(Object obj) {
-      if (!(obj instanceof MethodDef)) return false;
-      MethodDef md = (MethodDef) obj;
-      return equals(md.name, md.arg_types);
-    }
+  /** Map from a method to its replacement. */
+  private final Map<MethodDef, MethodDef> replacementMap;
 
-    @Override
-    public int hashCode() {
-      int code = name.hashCode();
-      for (Type arg : arg_types) code += arg.hashCode();
-      return code;
-    }
+  /** Maps a class name prefix to its replacement. */
```

Comment:
This code is now gone, but I've replaced the idea of a prefix by just saying "class or package name"